numba/tests/test_np_functions.py

```diff
@@ -2503,6 +2507,43 @@ def check_pass_through(jitted, expect_same, params):
                 else:
                     check_pass_through(cfunc, True, params)
 
+    def test_repeat(self):
+        pyfunc = np_repeat
+        cfunc = jit(nopython=True)(pyfunc)
+
+        def check(a, repeats):
+            np.testing.assert_equal(pyfunc(a, repeats), cfunc(a, repeats))
+
+        target_inputs = [
+            np.ones(1),
+            np.arange(1000),
+            np.array([[0, 1], [2, 3]]),
+            np.array([]),
+            np.array([[], []]),
+            1,
+            1.0,
+            True,
+            1j,
+            [0, 1, 2],
+            (0, 1, 2),
+        ]
+        for i in target_inputs:
+            check(i, repeats=0)
+            check(i, repeats=1)
+            check(i, repeats=2)
+
+    def test_repeat_exception(self):
+        pyfunc = np_repeat
+        cfunc = jit(nopython=True)(pyfunc)
+        self.disable_leak_check()
+
+        with self.assertRaises(ValueError):
```

Comment:
A typical pattern when checking that an error is raised is to check the type of the error is correct, but also check that the error message contains something expected, this is to ensure that the error caught is the one that was expected to be caught. This is a common pattern:
```python
with self.assertRaises(<some error class>) as e:
   <thing that causes error>
with self.assertIn("expected message", str(e.exception))
```